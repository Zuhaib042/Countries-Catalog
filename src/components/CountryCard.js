import React from 'react';
import { BsArrowRightCircle } from 'react-icons/bs';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

const CountryCard = ({ id, name, region, flagImage, population }) => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate(`/countrydetails/${id}`);
  };

  const convertNumber = (labelValue) => {
    if (Math.abs(Number(labelValue)) >= 1.0e9) {
      return `${(Math.abs(Number(labelValue)) / 1.0e9).toFixed(2)}B`;
    }
    if (Math.abs(Number(labelValue)) >= 1.0e6) {
      return `${(Math.abs(Number(labelValue)) / 1.0e6).toFixed(2)}M`;
    }
    if (Math.abs(Number(labelValue)) >= 1.0e3) {
      return `${(Math.abs(Number(labelValue)) / 1.0e3).toFixed(2)}K`;
    }
    return Math.abs(Number(labelValue));
  };
  const refined = convertNumber(population);

  return (
    <li className="col-span-1 bg-light p-3 text-white">
      <div className="flex justify-between">
        <div>
          <p className="text-2xl py-1">{name}</p>
          <p className="font-bold">{region}</p>
        </div>
        <div className="w-24 h-24">
          <img src={flagImage} width="100%" height="100%" alt="country-flag" />
        </div>
      </div>
      <div className="flex justify-between items-center">
        <p className="mt-5">
          <span className=""> Population: </span>
          <span className="text-2xl font-bold">{refined}</span>
        </p>
        <button onClick={handleClick} type="button" className="text-2xl">
          <BsArrowRightCircle />
        </button>
      </div>
    </li>
  );
};

CountryCard.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  region: PropTypes.string.isRequired,
  flagImage: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
};

export default CountryCard;
