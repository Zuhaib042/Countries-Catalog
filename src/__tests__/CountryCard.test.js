import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import CountryCard from '../components/CountryCard';

describe('Navbar test', () => {
  const country = {
    countryId: '12345',
    countryName: 'New Zealand',
    region: 'Africa',
    flag: '',
    population: 456778,
  };
  it('should render header correctly', () => {
    const missionsPage = render(
      <Provider store={store}>
        <Router>
          <CountryCard
            key={country.countryId}
            name={country.countryName}
            id={country.countryId}
            region={country.region}
            flagImage={country.flag}
            population={country.population}
          />
        </Router>
      </Provider>,
    );
    expect(missionsPage).toMatchSnapshot();
  });
});
