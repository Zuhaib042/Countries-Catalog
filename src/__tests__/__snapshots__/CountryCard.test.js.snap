// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test should render header correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="col-span-1 bg-light p-3 text-white"
      >
        <div
          class="flex justify-between"
        >
          <div>
            <p
              class="text-2xl py-1"
            >
              New Zealand
            </p>
            <p
              class="font-bold"
            >
              Africa
            </p>
          </div>
          <div
            class="w-24 h-24"
          >
            <img
              alt="country-flag"
              height="100%"
              src=""
              width="100%"
            />
          </div>
        </div>
        <div
          class="flex justify-between items-center"
        >
          <p
            class="mt-5"
          >
            <span
              class=""
            >
               Population: 
            </span>
            <span
              class="text-2xl font-bold"
            >
              456.78K
            </span>
          </p>
          <button
            class="text-2xl"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="col-span-1 bg-light p-3 text-white"
    >
      <div
        class="flex justify-between"
      >
        <div>
          <p
            class="text-2xl py-1"
          >
            New Zealand
          </p>
          <p
            class="font-bold"
          >
            Africa
          </p>
        </div>
        <div
          class="w-24 h-24"
        >
          <img
            alt="country-flag"
            height="100%"
            src=""
            width="100%"
          />
        </div>
      </div>
      <div
        class="flex justify-between items-center"
      >
        <p
          class="mt-5"
        >
          <span
            class=""
          >
             Population: 
          </span>
          <span
            class="text-2xl font-bold"
          >
            456.78K
          </span>
        </p>
        <button
          class="text-2xl"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
